ogr/abstract.py

```diff
@@ -726,6 +727,13 @@ def project_create(
         """
         raise NotImplementedError()
 
+    @property
+    def get_hostname(self, url: str) -> str:
+        repo_url = parse_git_repo(potential_url=url)
+        service_repo_url = parse_git_repo(repo_url.get_instance_url())
+        hostname = service_repo_url.hostname
+        return hostname
```

Comment:
1. The properties cannot have an argument since they are used without parenthesis (e.g. `my_service.hostname`).
```suggestion
    @property
    def get_hostname(self) -> str:
        repo_url = parse_git_repo(potential_url=url)
        service_repo_url = parse_git_repo(repo_url.get_instance_url())
        hostname = service_repo_url.hostname
        return hostname
```

2. Use `self.instance_url` instead:

```suggestion
    @property
    def get_hostname(self) -> str:
        repo_url = parse_git_repo(potential_url=self.instance_url)
        service_repo_url = parse_git_repo(repo_url.get_instance_url())
        hostname = service_repo_url.hostname
        return hostname
```

3. The properties are used as attributes => the `get` does not make sense here:

```suggestion
    @property
    def hostname(self) -> str:
        repo_url = parse_git_repo(potential_url=self.instance_url)
        service_repo_url = parse_git_repo(repo_url.get_instance_url())
        hostname = service_repo_url.hostname
        return hostname
```


3.  We don't need to parse the url twice:

```suggestion
    @property
    def hostname(self) -> str:
        service_repo_url = parse_git_repo(potential_url=self.instance_url)
        hostname = service_repo_url.hostname
        return hostname
```